processor/processor.go

```diff
@@ -1000,9 +995,22 @@ func (proc *HandleT) handlePendingGatewayJobs() bool {
 	proc.logger.Debugf("Processor DB Read size: %v", toQuery)
 	//Should not have any failure while processing (in v0) so
 	//retryList should be empty. Remove the assert
-	retryList := proc.gatewayDB.GetToRetry([]string{gateway.CustomVal}, toQuery, nil)
-	toQuery -= len(retryList)
-	unprocessedList := proc.gatewayDB.GetUnprocessed([]string{gateway.CustomVal}, toQuery, nil)
+
+	var retryList, unprocessedList []*jobsdb.JobT
+	var totalRetryEvents, totalUnprocessedEvents int
+	var gotMaxEvents bool
+
+	unTruncatedRetryList := proc.gatewayDB.GetToRetry([]string{gateway.CustomVal}, toQuery, nil)
+	retryList, totalRetryEvents, gotMaxEvents = getTruncatedEventList(unTruncatedRetryList, maxEventsToProcess)
```

Comment:
```
retryList, totalRetryEvents = getTruncatedEventList(unTruncatedRetryList, maxEventsToProcess)
eventsLeftToProcess := maxEventsToProcess - totalRetryEvents

if len(unTruncatedRetryList) >= dbReadBatchSize || eventsLeftToProcess <= 0 {
```

Reply:
@SumanthPuram  suppose maxEvents is 1000 and totalRetryEvents is 950 although there are more retryJobs left (eg. next retry job has 100 events, including it in retryJobs will make totalRetryEvents 1050 ( > 1000). This is the reason why gotMaxEvents is also returned